### PR TITLE
WIP: implement strict picking to support ignoring channels and clean grouping by type

### PR DESCRIPTION
**Pull request by dengemann**
The goals of this PR are at least threefold:

1. enable ignoring channels, e.g. EOG
2. ignore channels previously marked as bad, consistent with MNE-Python behavior
3. prepare clean grouping by channel types

Don't merge it is not yet done  and does not work. 
cc @jasmainak

### CODE REVIEW

**Comment by dengemann** [comment]
@jasmainak we also have some MNE API issues here. I think we should require the next release as minimum. See commits which remove `add_eeg_ref` keywords.

**Comment by dengemann** [comment]
@jasmainak feel free to add to this whenever you can. Maybe we can ratchet up some progress. I'll give you push access here.

File: autoreject/autoreject.py [hunk 1]
```diff
@@ -40,6 +43,12 @@ def _check_data(epochs):
                'incomplete data). Please check that no epoch '
                'is dropped when you call epochs.drop_bad_epochs().')
         raise RuntimeError(msg)
+    n_bads = len(epochs.info['bads'])
+    if n_bads > 0:
+        logger.info(
+            '%i channels are marked as bad. These will be ignored.'
```
Comment (agramfort) [inline autoreject/autoreject.py]: space missing after .

**Comment by agramfort** [comment]
do you treat bads as channels forced to be interpolated like maxfilter or ignored?

**Comment by jasmainak** [comment]
> do you treat bads as channels forced to be interpolated like maxfilter or ignored?

bads are ignored at the moment ...

**Comment by dengemann** [comment]
@agramfort we've been pair programming with @jasmainak :) we implemented a clean way of running auto reject on a subset of channels. Just cracked it, issue unblocked. Writing tests + updating examples later.

**Comment by agramfort** [comment]
cool !

**Comment by jasmainak** [comment]
@dengemann can you add tests? Then, we are good to merge

**Comment by dengemann** [comment]
Can you take over? I might or might not find the time soon.

**Comment by jasmainak** [comment]
okay cool, I will take over

File: autoreject/autoreject.py [hunk 2]
```diff
@@ -40,6 +43,16 @@ def _check_data(epochs):
...
+    n_bads = len(epochs.info['bads'])
+
+    if sum(ch_type in epochs for ch_type in ('mag', 'grad', 'eeg')) > 1:
+        raise ValueError('AutoReject handles only one channel type for now')
+
+    if n_bads > 0:
+        logger.info(
+            '%i channels are marked as bad. These will be ignored.'
+            'If you want them to be considered by autoreject please remove '
+            'them from epochs.info["bads"].' % n_bads)
```
Comment (dengemann) [inline autoreject/autoreject.py]: this needs a test

File: autoreject/autoreject.py [hunk 3]
```diff
@@ -590,23 +620,26 @@ def fit(self, epochs):
             The instance.
         """
         _check_data(epochs)
+        self.picks = _handle_picks(info=epochs.info, picks=self.picks)
```
Comment (dengemann) [inline autoreject/autoreject.py]: this attribute should also be tested

File: autoreject/autoreject.py [hunk 4]
```diff
@@ -412,7 +431,9 @@ def fit(self, epochs):
             The epochs object from which the channel-level thresholds are
             estimated.
         """
-        self.threshes_ = self.thresh_func(epochs, verbose=self.verbose)
+        self.picks = _handle_picks(info=epochs.info, picks=self.picks)
+        self.threshes_ = self.thresh_func(
```
Comment (dengemann) [inline autoreject/autoreject.py]: this .threshes_ should also be tested

File: autoreject/tests/test_viz.py [hunk 5]
```diff
@@ -34,6 +34,7 @@ def test_viz():
     n_epochs, n_channels, _ = epochs.get_data().shape
     fix_log = np.zeros((n_epochs, n_channels))
 
+    epochs.name = ''
```
Comment (jasmainak) [inline autoreject/tests/test_viz.py]: @dengemann I couldn't get this to work. This is a hack

**Comment by jasmainak** [comment]
@dengemann I added some tests. Take a look when you have time

**Comment by dengemann** [comment]
Here we go. Thanks for this nice peer/extreme programming & team effort @jasmainak

**Comment by jasmainak** [comment]
Thanks @dengemann ! And coverage is now up to 88% :)

**Pull request merged**